src/history/hs_verify.c

```diff
@@ -75,7 +75,7 @@ __hs_verify_id(
             F_SET(S2C(session), WT_CONN_DATA_CORRUPTION);
             WT_ERR_PANIC(session, WT_PANIC,
               "the associated history store key %s was not found in the data store %s",
-              __wt_buf_set_printable(session, key.data, key.size, false, prev_key),
+              __wt_key_string(session, key.data, key.size, S2BT(session)->key_format, &key),
```

Comment:
I suggest using `CUR2BT(ds_cbt)` instead of `S2BT` as we may change later that session may point to a different handle and it may break.

Also `key` is pointing to the history store key buffer. Better to use the `prev_key` that is a local buffer.

Reply:
Thanks, using CUR2BT here: 9abe66d8968e73b4042442916d8b596463d8435e.